libkbfs/tlf_handle.go

```diff
@@ -997,3 +534,52 @@ func (h TlfHandle) IsFinal() bool {
 func (h TlfHandle) IsConflict() bool {
 	return h.conflictInfo != nil
 }
+
+// GetPreferredFormat returns a TLF name formatted with the username given
+// as the parameter first.
+// This calls FavoriteNameToPreferredTLFNameFormatAs with the canonical
+// tlf name which will be reordered into the preferred format.
+// An empty username is allowed here and results in the canonical ordering.
+func (h TlfHandle) GetPreferredFormat(username libkb.NormalizedUsername) string {
+	s, err := FavoriteNameToPreferredTLFNameFormatAs(
+		username, h.GetCanonicalName())
+	if err != nil {
+		panic("TlfHandle.GetPreferredFormat: Parsing canonical username failed!")
+	}
+	return s
+}
+
+// FavoriteNameToPreferredTLFNameFormatAs formats a favorite names for display with the
+// username given.
+// An empty username is allowed here and results in tlfname being returned unmodified.
+func FavoriteNameToPreferredTLFNameFormatAs(username libkb.NormalizedUsername,
+	canon CanonicalTlfName) (string, error) {
+	tlfname := string(canon)
+	if len(username) == 0 {
+		return tlfname, nil
+	}
+	ws, rs, ext, err := splitTLFName(tlfname)
+	if err != nil {
+		return "", err
+	}
+	if len(ws) == 0 {
+		return "", fmt.Errorf("TLF name %q with no writers", tlfname)
+	}
+	uname := username.String()
+	for i, w := range ws {
+		if i != 0 && w == uname {
+			ws0 := ws[i]
```

Comment:
Done, was just writing the slice sliding code while ignoring the context with w already there.